src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -445,7 +443,8 @@ bool layout::compatible(const layout& other) const {
         return false;
     if (l1.get_linear_size() != l2.get_linear_size())
         return false;
-
+    if (check_redundant_1d_along_feature(l1, l2))
+        return true;
```

Comment:
this looks like a bug in this function. In such test case of os_iyx_osv16, this function should not return true.

Reply:
os_iyx_osv16 has block axis=0 and it consumes more memory that other block axis=1 formats for plain formats. So checking get_linear_size() is required. I updated the function. Thanks!